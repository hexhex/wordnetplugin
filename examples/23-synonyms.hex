%% this simply returns all synonyms of a word
%% in all parts of speech

%% the search number for synonyms is 23

word("work").
synonyms(W,SW,SY,P) :- &wordnet[23,W](P,SW,SY,_),word(W).

synonyms("work",1,"work",2), 
synonyms("work",2,"work",1), 
synonyms("work",2,"work",2), 
synonyms("work",3,"act",2), 
synonyms("work",3,"employment",1), 
synonyms("work",3,"work",1), 
synonyms("work",3,"work",2), 
synonyms("work",4,"operate",2), 
synonyms("work",4,"function",2), 
synonyms("work",4,"study",1), 
synonyms("work",4,"work",1)?

