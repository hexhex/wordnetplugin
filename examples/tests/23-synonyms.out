{synonyms("work",1,"work",1), synonyms("work",1,"work",2), synonyms("work",2,"do_work",2), synonyms("work",2,"piece_of_work",1), synonyms("work",2,"work",1), synonyms("work",2,"work",2), synonyms("work",3,"act",2), synonyms("work",3,"employment",1), synonyms("work",3,"work",1), synonyms("work",3,"work",2), synonyms("work",4,"run",2), synonyms("work",4,"go",2), synonyms("work",4,"operate",2), synonyms("work",4,"function",2), synonyms("work",4,"study",1), synonyms("work",4,"work",1), synonyms("work",4,"work",2), synonyms("work",5,"process",2), synonyms("work",5,"work_on",2), synonyms("work",5,"work",1), synonyms("work",5,"work",2), synonyms("work",6,"work_out",2), synonyms("work",6,"exercise",2), synonyms("work",6,"workplace",1), synonyms("work",6,"work",1), synonyms("work",6,"work",2), synonyms("work",7,"make",2), synonyms("work",7,"body_of_work",1), synonyms("work",7,"oeuvre",1), synonyms("work",7,"work",1), synonyms("work",7,"work",2), synonyms("work",8,"work",2), synonyms("work",9,"work",2), synonyms("work",10,"work",2), synonyms("work",11,"make_for",2), synonyms("work",11,"wreak",2), synonyms("work",11,"play",2), synonyms("work",11,"bring",2), synonyms("work",11,"work",2), synonyms("work",12,"put_to_work",2), synonyms("work",12,"work",2), synonyms("work",13,"crop",2), synonyms("work",13,"cultivate",2), synonyms("work",13,"work",2), synonyms("work",14,"work",2), synonyms("work",15,"act_upon",2), synonyms("work",15,"influence",2), synonyms("work",15,"work",2), synonyms("work",16,"work",2), synonyms("work",17,"work",2), synonyms("work",18,"work",2), synonyms("work",19,"work",2), synonyms("work",20,"forge",2), synonyms("work",20,"mould",2), synonyms("work",20,"mold",2), synonyms("work",20,"form",2), synonyms("work",20,"shape",2), synonyms("work",20,"work",2), synonyms("work",21,"work",2), synonyms("work",22,"knead",2), synonyms("work",22,"work",2), synonyms("work",23,"exploit",2), synonyms("work",23,"work",2), synonyms("work",24,"lick",2), synonyms("work",24,"puzzle_out",2), synonyms("work",24,"figure_out",2), synonyms("work",24,"solve",2), synonyms("work",24,"work_out",2), synonyms("work",24,"work",2), synonyms("work",25,"ferment",2), synonyms("work",25,"work",2), synonyms("work",26,"turn",2), synonyms("work",26,"sour",2), synonyms("work",26,"ferment",2), synonyms("work",26,"work",2), synonyms("work",27,"work",2), word("work")}
